### ISSUE TITLE
Unable to parse access using an interpolation containing an access

### ISSUE THREAD

**Issue by lollipopman** [puppetlabs/puppet-lint#267]
## Describe the Bug
parsing:
```puppet
$foo["${facts['networking']['hostname']}"]
```
fails with:
```
ERROR: Syntax error (unterminated string) on line 1 (check: syntax)
```

## Expected Behavior
no error, parses fine with `puppet-parser -v`
